internal/addrs/remove_endpoint.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package addrs
+
+import (
+	"github.com/hashicorp/hcl/v2"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+// RemoveEndpoint is to ConfigMoveable what Target is to Targetable:
+// a wrapping struct that captures the result of decoding an HCL
+// traversal representing a relative path from the current module to
+// a removable object. It is very similar to MoveEndpoint.
+//
+// Its purpose is for the "from" address in a "removed" statement
+// in the configuration.
+//
+// To obtain a full address from a RemoveEndpoint we need to combine it
+// with any ancestor modules in the configuration
+type RemoveEndpoint struct {
+	// SourceRange is the location of the physical endpoint address
+	// in configuration, if this RemoveEndpoint was decoded from a
+	// configuration expression.
+	SourceRange tfdiags.SourceRange
+
+	// the representation of our relative address as a ConfigRemovable
+	RelSubject ConfigRemovable
+}
+
+func (e *RemoveEndpoint) String() string {
+	// Our internal pseudo-AbsMoveable representing the relative
+	// address (either ModuleInstance or AbsResourceInstance) is
+	// a good enough proxy for the relative move endpoint address
+	// serialization.
```

Comment:
this comment seems to be for move stuff, not remove. 

is it okay for us to re-use this comment here? 

Also, do we need to implement a psuedo-AbsRemovable in module_call.go?